meta.go

```diff
@@ -122,6 +122,44 @@ type Storage struct {
 	Properties []string `bson:"properties,omitempty"`
 }
 
+// DeviceType defines a device type
+type DeviceType string
+
+// DeviceGPU represents GPU device type
+const DeviceGPU DeviceType = "gpu"
+
+// DeviceTypes defines current supported device types.
+var DeviceTypes = []DeviceType{DeviceGPU}
+
+// Validate does validation check for DeviceType
+func (dt *DeviceType) Validate() error {
+	for _, t := range DeviceTypes {
+		if *dt == t {
+			return nil
+		}
+	}
+	return fmt.Errorf("Invalid device type: %q, current supported are %+v", dt, DeviceTypes)
+}
+
+// Device represents a charm's device requirement(GPU for example).
+type Device struct {
+	// Name is the name of the device.
+	// name of the GPU device on k8s could be like: nvidia.com/gpu
+	Name string `bson:"name"`
+
+	// Description is a description of the device.
+	Description string `bson:"description"`
+
+	// Type is the device type
+	Type DeviceType `bson:"type"`
+
+	// Request is the min number of devices to use
+	Request int64 `bson:"request"`
```

Comment:
Instead of Request and Limit, we should stick to the convention used by Storage, CountMin and CountMax

Reply:
I was trying to use k8s resource limit/request naming. But u r right, it should be consistent with Storage. updated